src/HealthGPS/energy_balance_hierarchical_model.cpp

```diff
@@ -1,18 +1,18 @@
 #include "energy_balance_hierarchical_model.h"
 #include "runtime_context.h"
-#include <iostream>
 
 namespace hgps {
 
 EnergyBalanceHierarchicalModel::EnergyBalanceHierarchicalModel(
-    LiteHierarchicalModelDefinition &definition)
-    : definition_{definition} {}
+    const std::map<core::IntegerInterval, AgeGroupGenderEquation> &equations,
+    const std::map<core::Identifier, core::Identifier> &variables, double boundary_percentage)
+    : equations_{equations}, variables_{variables}, boundary_percentage_{boundary_percentage} {}
 
 HierarchicalModelType EnergyBalanceHierarchicalModel::type() const noexcept {
     return HierarchicalModelType::Dynamic;
 }
 
-const std::string &EnergyBalanceHierarchicalModel::name() const noexcept { return name_; }
+const std::string EnergyBalanceHierarchicalModel::name() const noexcept { return "Dynamic"; }
```

Comment:
Won't a reference to const string be optimised away anyway?

Reply:
Oh wait, I was being daft. It shouldn't be a reference because you're creating a new ``std::string`` (from the `const char*`). My bad.

That said, if it's not a reference it doesn't need to be `const`. The caller can then decide whether they want it const or not.

Reply:
Ah, true.